internal/services/controller/controller.go

```diff
@@ -535,21 +535,40 @@ func getConditionalInformers(f informers.SharedInformerFactory, df dynamicinform
 	}
 }
 
-func getDefaultInformers(f informers.SharedInformerFactory) map[reflect.Type]cache.SharedInformer {
-	return map[reflect.Type]cache.SharedInformer{
-		reflect.TypeOf(&corev1.Node{}):                  f.Core().V1().Nodes().Informer(),
-		reflect.TypeOf(&corev1.Pod{}):                   f.Core().V1().Pods().Informer(),
-		reflect.TypeOf(&corev1.PersistentVolume{}):      f.Core().V1().PersistentVolumes().Informer(),
-		reflect.TypeOf(&corev1.PersistentVolumeClaim{}): f.Core().V1().PersistentVolumeClaims().Informer(),
-		reflect.TypeOf(&corev1.ReplicationController{}): f.Core().V1().ReplicationControllers().Informer(),
-		reflect.TypeOf(&corev1.Namespace{}):             f.Core().V1().Namespaces().Informer(),
-		reflect.TypeOf(&corev1.Service{}):               f.Core().V1().Services().Informer(),
-		reflect.TypeOf(&appsv1.Deployment{}):            f.Apps().V1().Deployments().Informer(),
-		reflect.TypeOf(&appsv1.ReplicaSet{}):            f.Apps().V1().ReplicaSets().Informer(),
-		reflect.TypeOf(&appsv1.DaemonSet{}):             f.Apps().V1().DaemonSets().Informer(),
-		reflect.TypeOf(&appsv1.StatefulSet{}):           f.Apps().V1().StatefulSets().Informer(),
-		reflect.TypeOf(&storagev1.StorageClass{}):       f.Storage().V1().StorageClasses().Informer(),
-		reflect.TypeOf(&batchv1.Job{}):                  f.Batch().V1().Jobs().Informer(),
+type defaultInformer struct {
+	informer cache.SharedInformer
+	filters  filters.Filters
+}
+
+func getDefaultInformers(f informers.SharedInformerFactory) map[reflect.Type]defaultInformer {
```

Comment:
can we use the opportunity to change this to `map[string]defaultInformer` ? i don't think we ever use the reflect.Type other than immediately converting it to string.